recipes/dgl/build.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -euxo pipefail
+
+rm -rf build || true
+
+if [ ${cuda_compiler_version} != "None" ]; then
+CUDA_CMAKE_OPTIONS="-DCMAKE_CUDA_COMPILER=${CUDA_HOME}/bin/nvcc -DCMAKE_CUDA_HOST_COMPILER=${CXX}"
+USE_CUDA=ON
+else
+CUDA_CMAKE_OPTIONS=""
+USE_CUDA=OFF
+fi
+
+# Remove -std=c++17 from CXXFLAGS for compatibility with nvcc
+export CXXFLAGS="$(echo $CXXFLAGS | sed -e 's/ -std=[^ ]*//')"
+export CFLAGS="$(echo $CFLAGS | sed -e 's/ -mtune=[^ ]*//')"
+CMAKE_FLAGS="${CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=${PREFIX} -DCMAKE_BUILD_TYPE=Release -DPython_EXECUTABLE=${PYTHON}"
+
```

Comment:
```suggestion
if [[ ${cuda_compiler_version} != "None" ]]; then
    if [[ ${cuda_compiler_version} == 9.0* ]]; then
        export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;7.0+PTX"
    elif [[ ${cuda_compiler_version} == 9.2* ]]; then
        export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0+PTX"
    elif [[ ${cuda_compiler_version} == 10.* ]]; then
        export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5+PTX"
    elif [[ ${cuda_compiler_version} == 11.0* ]]; then
        export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0+PTX"
    elif [[ ${cuda_compiler_version} == 11.1 ]]; then
        export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
    elif [[ ${cuda_compiler_version} == 11.2 ]]; then
        export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
    else
        echo "unsupported cuda version. edit build.sh"
        exit 1
    fi
fi
```
Please target the [same archs as the pytorch package](https://github.com/conda-forge/pytorch-cpu-feedstock/blob/229b44730ae4e9b1cb144910f0a3a25cee0e7391/recipe/build_pytorch.sh#L128C32-L128C32). It looks like you're targeting [3.5,5.0,8.0,8.6+PTX](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=753751&view=logs&j=eb6ecf2b-0363-5c1a-baa7-2d771889d197&t=b342d271-df1b-539d-7106-021ce5ffca81&l=2418), which probably doesn't make sense because the package wouldn't run on any 6.x,7.x devices.

Reply:
We can try this but we might run out of disk space/system memory building all of these different arches

Reply:
If you run out of space, then use two strategies:

1. build for major versions only: 3.5;5.0;6.0;7.0;8.0+PTX
2. Cut newer versions: 3.5;5.0;6.0;7.0+PTX

Reply:
`All` looks pretty close to this logic 
https://github.com/dmlc/dgl/blob/master/cmake/modules/CUDA.cmake#L13-L26